crates/turbopack-cli/src/dev/turbo_tasks_viz.rs

```diff
@@ -30,12 +32,51 @@ const INVALIDATION_INTERVAL: Duration = Duration::from_secs(3);
 
 #[turbo_tasks::value_impl]
 impl ContentSource for TurboTasksSource {
+    #[turbo_tasks::function]
+    fn get_routes(self_vc: TurboTasksSourceVc) -> RouteTreeVc {
+        RouteTreesVc::cell(vec![
```

Comment:
Do we need a `RouteTreesVc`? Could we do with just `RouteTreeVc::merge` directly instead?

Reply:
Yes, we need that. I had `RouteTreeVc::merge` before, but this has weird effects.

`Vec<RouteTreeVc>` as argument creates a new instance for every different list of RouteTrees.
`RouteTreesVc::cell(...).merge()` stores the list of routes in a cell and the merge function is always the same instance, so it the resulting RouteTreeVc.
So "celling" the Vec of RouteTrees can reuse the tasks and might end the merging early when the resulting RouteTree is equal (flat equal!).

Reply:
Can you add a comment to the `merge` function to explain this?

Since this is a more turbo-tasks specific pattern, it would be helpful to add this case to some "recommended patterns" document.